google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -313,4 +326,30 @@ public void testGetServiceAccountProjectId_nonExistentFile() throws Exception {
 
     assertNull(ServiceOptions.getServiceAccountProjectId(credentialsFile.getPath()));
   }
+
+  @Test
+  public void testResponseHeaderContainsMetaDataFlavor() throws Exception {  
+    // HttpTransport transport = new MockHttpTransport() {
+    //   @Override
+    //   public LowLevelHttpRequest buildRequest(String method, String url) throws IOException {
+    //     return new MockLowLevelHttpRequest() {
+    //       @Override
+    //       public LowLevelHttpResponse execute() throws IOException {
+    //         MockLowLevelHttpResponse response = new MockLowLevelHttpResponse();
+    //         response.addHeader("Metadata-Flavor", "Google");
+    //         response.setStatusCode(200);
+    //         return response;
+    //       }
+    //     };
+    //   }
+    // };
+    HttpTransport mockHttpTransport = Mockito.mock(MockHttpTransport.class);
+    MockLowLevelHttpResponse mockResponse = new MockLowLevelHttpResponse();
+    mockResponse.addHeader("Metadata-Flavor", "Google");
+    Mockito.when(mockHttpTransport.buildRequest(Matchers.anyString(), Matchers.anyString()).execute()).thenReturn(mockResponse);
```

Comment:
This won't work, due to the fact that `buildRequest(String, String)` has protected access in `com.google.api.client.http.HttpTransport`

Reply:
https://developers.google.com/api-client-library/java/google-http-java-client/reference/1.20.0/com/google/api/client/testing/http/MockHttpTransport to the rescue!